tests/functional/object/mcg/test_lifecycle_configuration.py

```diff
@@ -0,0 +1,365 @@
+        # older versions + newer versions + the current version
+        amount = 2 * older_versions_amount + 1
+        upload_obj_versions(
+            mcg_obj,
+            awscli_pod,
+            bucket,
+            key,
+            amount=amount,
+        )
+
+        # 4. Manually set the age of each version to be one day older than its successor
+        uploaded_versions = get_obj_versions(mcg_obj, awscli_pod, bucket, key)
+        version_ids = [version["VersionId"] for version in uploaded_versions]
+
+        # Parse the timestamp from the first version
+        mongodb_style_time = uploaded_versions[0]["LastModified"]
+        iso_timestamp = mongodb_style_time.replace("Z", "+00:00")
+        latest_version_creation_date = datetime.fromisoformat(iso_timestamp)
+
+        for i, version_id in enumerate(version_ids):
+            change_versions_creation_date_in_noobaa_db(
+                bucket_name=bucket,
+                object_key=key,
+                version_ids=[version_id],
+                new_creation_time=(
+                    latest_version_creation_date - timedelta(days=i)
+                ).timestamp(),
+            )
+
+        # 5. Wait for versions to expire
+        # While older_versions_amount versions qualify for deletion due to
+        # NoncurrentDays, the lifecycle policy should keep the NewerNoncurrentVersions
+        # amount of versions.
+        expected_remaining = set(
+            version_ids[: newer_versions_amount + 1]
+        )  # +1 for the current version
+
+        for versions in TimeoutSampler(
+            timeout=TIMEOUT_THRESHOLD,
+            sleep=TIMEOUT_SLEEP_DURATION,
+            func=get_obj_versions,
+            mcg_obj=mcg_obj,
+            awscli_pod=awscli_pod,
+            bucket_name=bucket,
+            obj_key=key,
+        ):
+            remaining = {v["VersionId"] for v in versions}
+
+            # Expected end result
+            if remaining == expected_remaining:
+                logger.info("Only the expected versions remained")
+                break
+
+            # Newer versions were deleted
+            elif not (expected_remaining <= remaining):
+                raise UnexpectedBehaviour(
+                    (
+                        "Some versions were deleted when they shouldn't have!"
+                        f"Versions that were deleted: {expected_remaining - remaining}"
+                    )
+                )
+
+            # Some older versions are yet to be deleted
+            else:
+                logger.warning(
+                    (
+                        "Some older versions have not expired yet:\n"
+                        f"Remaining: {remaining}\n"
+                        f"Versions yet to expire: {remaining - expected_remaining}"
+                    )
+                )
+
+    @tier1
+    @pytest.mark.polarion_id("OCS-6802")
+    def test_expired_object_delete_marker(self, mcg_obj, bucket_factory, awscli_pod):
+        """
```

Comment:
Add args in Docstring

Reply:
We typically don't include fixture args in docstrings for test functions, so I think it's best to keep it consistent with the rest of the project.